covariance/covarianceBase.cpp

```diff
@@ -444,7 +444,7 @@ void covarianceBase::init(std::vector<std::string> YAMLFile)
     }
   }
 
-  //setName(YAMLFile[0].c_str());
+  setName("yaml_systs");
```

Comment:
Why was this changed?
This in principle can lead to situation where two covariances have the same name which is not good.
If you don't want name of covariance to be driven by the name of yaml you can change constructor to:
```
covarianceBase::covarianceBase(std::vector<std::string> YAMLFile, std::string name)
setName(name);
```

Reply:
Good point.

I think setting name via constructor would be safest.
You can set default value to reduce potentail compilation error. But will leave it to up to you.